front/lib/api/assistant/agent/agent_get.ts

```diff
@@ -0,0 +1,195 @@
+import { Op } from "sequelize";
+
+import { Authenticator } from "@app/lib/auth";
+import { DataSource, Workspace } from "@app/lib/models";
+import {
+  AgentDataSourceConfiguration,
+  AgentRetrievalConfiguration,
+} from "@app/lib/models/assistant/actions/retrieval";
+import {
+  AgentConfiguration,
+  AgentGenerationConfiguration,
+} from "@app/lib/models/assistant/agent";
+import {
+  RetrievalDataSourcesConfiguration,
+  RetrievalQuery,
+  RetrievalTimeframe,
+} from "@app/types/assistant/actions/retrieval";
+import {
+  AgentActionConfigurationType,
+  AgentConfigurationType,
+  AgentGenerationConfigurationType,
+  AgentType as AgentType,
+} from "@app/types/assistant/agent";
+
+/**
+ * Get an agent full configuration from its name
+ */
+export async function getAgent(
+  auth: Authenticator,
+  sId: string
+): Promise<AgentType> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Cannot find Agent: no workspace");
+  }
+  const agent = await AgentConfiguration.findOne({
+    where: {
+      sId: sId,
+      workspaceId: owner.id,
+    },
+  });
+  if (!agent) {
+    throw new Error("Cannot find Agent: no workspace");
+  }
+  const agentGeneration = await AgentGenerationConfiguration.findOne({
+    where: {
+      agentId: agent.id,
+    },
+  });
+  const agentAction = await AgentRetrievalConfiguration.findOne({
+    where: {
+      agentId: agent.id,
+    },
+  });
+  const agentDataSources = agentAction?.id
+    ? await AgentDataSourceConfiguration.findAll({
+        where: {
+          retrievalConfigurationId: agentAction?.id,
+        },
+      })
+    : [];
+
+  return {
+    agent: await _buildAgentConfigurationTypeFromModel({ agent }),
+    action: agentAction
+      ? await _buildAgentActionConfigurationTypeFromModel(
+          agentAction,
+          agentDataSources || []
+        )
+      : null,
+    generation: agentGeneration
+      ? _buildAgentGenerationConfigurationTypeFromModel(agentGeneration)
+      : null,
+  };
+}
+
+/**
+ * Builds the agent configuration type from the model
+ */
+export async function _buildAgentConfigurationTypeFromModel({
+  agent,
+}: {
+  agent: AgentConfiguration;
+}): Promise<AgentConfigurationType> {
+  return {
+    id: agent.id,
+    sId: agent.sId,
+    name: agent.name,
+    pictureUrl: agent.pictureUrl,
+    status: agent.status,
+  };
+}
+
+/**
+ * Builds the agent generation configuration type from the model
+ */
+export function _buildAgentGenerationConfigurationTypeFromModel(
```

Comment:
really don't think we need that function makes the code harder to understand and does not really save much?